proof/refine/ARM_HYP/VSpace_R.thy

```diff
@@ -4513,8 +4513,15 @@ lemma storePDE_ifunsafe [wp]:
 
 lemma storePDE_idle [wp]:
   "\<lbrace>valid_idle'\<rbrace> storePDE p pde \<lbrace>\<lambda>rv. valid_idle'\<rbrace>"
-  unfolding valid_idle'_def
-  by (rule hoare_lift_Pf [where f="ksIdleThread"]; wp)
+  apply (simp add: valid_idle'_def)
```

Comment:
The occurrences are across all the arches, with max three occurrences for one arch, and, given how Refine is set up currently, we need to define such a method for each arch. And the method needs to take a parameter telling it what to unfold. So, although I’ve written the method anyway, it is probably not as effort saving as it might appear, unfortunately. I am not sure about adding it to RISCV as there is only one occurrence, but I might as well.

Reply:
Probably easier if it is the same in all arches. If/when we arch split refine, we might then be able to pull them together into one definition.